pkg/k8shandler/fluentd.go

```diff
@@ -414,6 +414,38 @@ func newFluentdPodSpec(cluster *logging.ClusterLogging, elasticsearchAppName str
 	// Shorten the termination grace period from the default 30 sec to 10 sec.
 	fluentdPodSpec.TerminationGracePeriodSeconds = utils.GetInt64(10)
 
+	// FIXME: The following conditional branch is a refactoring candidate
+	// To address here https://issues.redhat.com/browse/LOG-833
+	//
+	// Why do we need this branch at all?
+	//
+	// The ClusterLogging CR supports three additional use cases without providing
+	// a logStore stanza namely:
+	// Case 1: A collection only stanza to use with secure forward
+	// Case 2: A collection only stanza to use with syslog
+	// Case 3: A collection only stanza to use with ClusterLogFowarder API
+	//
+	// Supporting all three cases implies:
+	// 1. We don't want to add the init container if `cluster.Spec.LogStore == nil`.
```

Comment:
"because spec.logstore==nil" means there is no elastic and thus there is no need to wait"?  (which covers case 1 and 2)?

Reply:
Yes this is correct.